File: apps/web/src/types/health.ts
export interface ServiceComponentHealth {
  status: string;
  connected: boolean;
  error?: string | null;
}

export interface DatabaseHealth extends ServiceComponentHealth {
  pgvector_installed: boolean;
}

export interface HealthResponse {
  status: 'healthy' | 'degraded' | 'unhealthy';
  project_name: string;
  version: string;
  environment: string;
  timestamp: string;
  database: DatabaseHealth;
  redis: ServiceComponentHealth;
}

File: apps/api/src/schemas/health.py
from datetime import datetime

from pydantic import BaseModel, Field


class ServiceComponentHealth(BaseModel):
    status: str = Field(..., examples=["healthy"])
    connected: bool = Field(..., examples=[True])
    error: str | None = None


class DatabaseHealth(ServiceComponentHealth):
    pgvector_installed: bool = Field(..., examples=[True])


class HealthResponse(BaseModel):
    status: str = Field(..., examples=["healthy"])
    project_name: str = Field(..., examples=["AI Knowledge Assistant"])
    version: str = Field(..., examples=["0.1.0"])
    environment: str = Field(..., examples=["development"])
    timestamp: datetime
    database: DatabaseHealth
    redis: ServiceComponentHealth


class ProbeResponse(BaseModel):
    status: str = Field(..., examples=["ok"])
    timestamp: datetime
